Instrumented remaining google deploy operations.

ewiseblatt opened the pull request:
@duftler 

I made these changes yesterday before your note about formatting. I chose to format the changes this way because I thought it was the most readable given the surrounding context. I dont mind going back and changing it since it is mechanical and have been through twice already and hasnt been a big deal.

The tests are slightly inconsistent. Sometimes I test for metrics and sometimes not. Where I dont, the registry scope doesnt really matter other than it needs to be defined. Where I do, I scope the registry to the test methods so that it wont be influenced elsewhere. I typically only test the main metric and typically only on success. I didnt want to make the tests too brittle because the metric lookup needs to have exact tags. But I wanted to call out differences between regional, zonal, global where the metric names differ. The successCode is "0" in the tests, even though that value doesnt make sense. That is a defincincy in the google client libraries. The runtime does give the right values (I tested this manually). I documented why each test was using "0".

duftler commented on clouddriver-google/src/main/groovy/com/netflix/spinnaker/clouddriver/google/deploy/ops/ModifyGoogleServerGroupInstanceTemplateAtomicOperation.groovy:
```diff
@@ -103,16 +102,23 @@ class ModifyGoogleServerGroupInstanceTemplateAtomicOperation implements AtomicOp
-      ? instanceGroupManagers.get(project, region, serverGroupName).execute()
-      : instanceGroupManagers.get(project, zone, serverGroupName).execute()
+      ? timeExecute(instanceGroupManagers.get(project, region, serverGroupName),
+                    "compute.regionalInstanceGroupManagers",
```
s/regional/region

duftler commented on clouddriver-google/src/main/groovy/com/netflix/spinnaker/clouddriver/google/deploy/ops/ModifyGoogleServerGroupInstanceTemplateAtomicOperation.groovy:
```diff
@@ -236,17 +245,23 @@ class ModifyGoogleServerGroupInstanceTemplateAtomicOperation implements AtomicOp
+        def setInstanceTemplateOperation = timeExecute(
+          instanceGroupManagers.setInstanceTemplate(
+            project, region, serverGroupName, regionInstanceGroupManagersSetTemplateRequest),
+          "compute.regionalInstanceGroupManagers.setInstanceTemplate",
```
s/regional/region

duftler commented on clouddriver-google/src/main/groovy/com/netflix/spinnaker/clouddriver/google/deploy/ops/UpsertGoogleServerGroupTagsAtomicOperation.groovy:
```diff
@@ -121,33 +132,46 @@ class UpsertGoogleServerGroupTagsAtomicOperation implements AtomicOperation<Void
+      def setInstanceTemplateOperation = timeExecute(
+              instanceGroupManagers.setInstanceTemplate(
+              project, region, serverGroupName, regionInstanceGroupManagersSetTemplateRequest),
+              "compute.regionalInstanceGroupManagers.setInstanceTemplate",
```
s/regional/region

duftler commented on clouddriver-google/src/main/groovy/com/netflix/spinnaker/clouddriver/google/deploy/ops/loadbalancer/DeleteGoogleSslLoadBalancerAtomicOperation.groovy:
```diff
@@ -142,7 +151,10 @@ class DeleteGoogleSslLoadBalancerAtomicOperation extends DeleteGoogleLoadBalance
-      { compute.backendServices().delete(project, backendServiceName).execute() },
+      { timeExecute(
+            compute.backendServices().delete(project, backendServiceName),
+            "compute.backendSerfvices.delete",
```
s/Serfvices/Services

duftler commented on clouddriver-google/src/main/groovy/com/netflix/spinnaker/clouddriver/google/deploy/ops/loadbalancer/UpsertGoogleSslLoadBalancerAtomicOperation.groovy:
```diff
@@ -201,7 +222,10 @@ class UpsertGoogleSslLoadBalancerAtomicOperation extends UpsertGoogleLoadBalance
-        { compute.backendServices().update(project, existingBackendService.getName(), existingBackendService).execute() },
+        { timeExecute(
+              compute.backendServices().update(project, existingBackendService.getName(), existingBackendService),
+              "compute.backendSerfvices.update",
```
s/Serfvices/Services

duftler commented on clouddriver-google/src/test/groovy/com/netflix/spinnaker/clouddriver/google/deploy/ops/DestroyGoogleServerGroupAtomicOperationUnitSpec.groovy:
```diff
@@ -234,6 +255,12 @@ class DestroyGoogleServerGroupAtomicOperationUnitSpec extends Specification {
         1 * zoneOperationsGet.execute() >> instanceGroupManagersDeleteOp
+        registry.timer(
+            registry.createId("google.api",
+                  [api: "compute.instanceGroupManagers.delete",
```
This map appears often enough in the tests that it might be worth it to create a helper method for the tests, something like: `Map generateMetricsTagsMap(String api, String scope)`

ewiseblatt commented:
Thanks for catching the typos. I wish the client library had the ability to
identify the requests once they've been created so I didnt need that
duplicate string.

I decided not to do the common tag map generator because there arguments
vary among instances. Some are region, some are zone. Though the common
method would only require 3 variables (api, region, zone) instead of 6, the
zone and region are disjoint so looks more awkward given there arent many
call sites.

On Mon, Jan 30, 2017 at 8:55 AM, Matt Duftler <notifications@github.com>
wrote:

> *@duftler* approved this pull request.
>
> Thanks for the detailed explanation.
> LGTM after addressing comments.
> ------------------------------
>
> In clouddriver-google/src/main/groovy/com/netflix/spinnaker/
> clouddriver/google/deploy/ops/ModifyGoogleServerGroupInstanc
> eTemplateAtomicOperation.groovy
> <https://github.com/spinnaker/clouddriver/pull/1403#pullrequestreview-19077911>
> :
>
> > @@ -103,16 +102,23 @@ class ModifyGoogleServerGroupInstanceTemplateAtomicOperation implements AtomicOp
>      // Retrieve the managed instance group.
>      def managedInstanceGroup =
>        isRegional
> -      ? instanceGroupManagers.get(project, region, serverGroupName).execute()
> -      : instanceGroupManagers.get(project, zone, serverGroupName).execute()
> +      ? timeExecute(instanceGroupManagers.get(project, region, serverGroupName),
> +                    "compute.regionalInstanceGroupManagers",
>
> s/regional/region
> ------------------------------
>
> In clouddriver-google/src/main/groovy/com/netflix/spinnaker/
> clouddriver/google/deploy/ops/ModifyGoogleServerGroupInstanc
> eTemplateAtomicOperation.groovy
> <https://github.com/spinnaker/clouddriver/pull/1403#pullrequestreview-19077911>
> :
>
> > @@ -236,17 +245,23 @@ class ModifyGoogleServerGroupInstanceTemplateAtomicOperation implements AtomicOp
>        if (isRegional) {
>          def regionInstanceGroupManagersSetTemplateRequest =
>            new RegionInstanceGroupManagersSetTemplateRequest(instanceTemplate: instanceTemplateUrl)
> -        def setInstanceTemplateOperation = instanceGroupManagers.setInstanceTemplate(
> -          project, region, serverGroupName, regionInstanceGroupManagersSetTemplateRequest).execute()
> +        def setInstanceTemplateOperation = timeExecute(
> +          instanceGroupManagers.setInstanceTemplate(
> +            project, region, serverGroupName, regionInstanceGroupManagersSetTemplateRequest),
> +          "compute.regionalInstanceGroupManagers.setInstanceTemplate",
>
> s/regional/region
> ------------------------------
>
> In clouddriver-google/src/main/groovy/com/netflix/spinnaker/
> clouddriver/google/deploy/ops/UpsertGoogleServerGroupTagsAto
> micOperation.groovy
> <https://github.com/spinnaker/clouddriver/pull/1403#pullrequestreview-19077911>
> :
>
> > @@ -121,33 +132,46 @@ class UpsertGoogleServerGroupTagsAtomicOperation implements AtomicOperation<Void
>      if (isRegional) {
>        def regionInstanceGroupManagersSetTemplateRequest =
>          new RegionInstanceGroupManagersSetTemplateRequest(instanceTemplate: instanceTemplateUrl)
> -      def setInstanceTemplateOperation = instanceGroupManagers.setInstanceTemplate(
> -        project, region, serverGroupName, regionInstanceGroupManagersSetTemplateRequest).execute()
> +      def setInstanceTemplateOperation = timeExecute(
> +              instanceGroupManagers.setInstanceTemplate(
> +              project, region, serverGroupName, regionInstanceGroupManagersSetTemplateRequest),
> +              "compute.regionalInstanceGroupManagers.setInstanceTemplate",
>
> s/regional/region
> ------------------------------
>
> In clouddriver-google/src/main/groovy/com/netflix/spinnaker/
> clouddriver/google/deploy/ops/loadbalancer/DeleteGoogleSslLoadBalancerAto
> micOperation.groovy
> <https://github.com/spinnaker/clouddriver/pull/1403#pullrequestreview-19077911>
> :
>
> > @@ -142,7 +151,10 @@ class DeleteGoogleSslLoadBalancerAtomicOperation extends DeleteGoogleLoadBalance
>      }
>
>      Operation deleteBackendServiceOp = GCEUtil.deleteIfNotInUse(
> -      { compute.backendServices().delete(project, backendServiceName).execute() },
> +      { timeExecute(
> +            compute.backendServices().delete(project, backendServiceName),
> +            "compute.backendSerfvices.delete",
>
> s/Serfvices/Services
> ------------------------------
>
> In clouddriver-google/src/main/groovy/com/netflix/spinnaker/
> clouddriver/google/deploy/ops/loadbalancer/UpsertGoogleSslLoadBalancerAto
> micOperation.groovy
> <https://github.com/spinnaker/clouddriver/pull/1403#pullrequestreview-19077911>
> :
>
> > @@ -201,7 +222,10 @@ class UpsertGoogleSslLoadBalancerAtomicOperation extends UpsertGoogleLoadBalance
>        existingBackendService.loadBalancingScheme = 'EXTERNAL'
>        existingBackendService.protocol = description.ipProtocol
>        backendServiceOp = safeRetry.doRetry(
> -        { compute.backendServices().update(project, existingBackendService.getName(), existingBackendService).execute() },
> +        { timeExecute(
> +              compute.backendServices().update(project, existingBackendService.getName(), existingBackendService),
> +              "compute.backendSerfvices.update",
>
> s/Serfvices/Services
> ------------------------------
>
> In clouddriver-google/src/test/groovy/com/netflix/spinnaker/
> clouddriver/google/deploy/ops/DestroyGoogleServerGroupAtomic
> OperationUnitSpec.groovy
> <https://github.com/spinnaker/clouddriver/pull/1403#pullrequestreview-19077911>
> :
>
> > @@ -234,6 +255,12 @@ class DestroyGoogleServerGroupAtomicOperationUnitSpec extends Specification {
>          1 * computeMock.zoneOperations() >> zoneOperations
>          1 * zoneOperations.get(PROJECT_NAME, location, INSTANCE_GROUP_OP_NAME) >> zoneOperationsGet
>          1 * zoneOperationsGet.execute() >> instanceGroupManagersDeleteOp
> +        registry.timer(
> +            registry.createId("google.api",
> +                  [api: "compute.instanceGroupManagers.delete",
>
> This map appears often enough in the tests that it might be worth it to
> create a helper method for the tests, something like: Map
> generateMetricsTagsMap(String api, String scope)
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/spinnaker/clouddriver/pull/1403#pullrequestreview-19077911>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AEKkGHC_6X8JsWi4ikHZkBTUJSzTmiAAks5rXevmgaJpZM4Lwnbi>
> .
>

ewiseblatt commented:
Maybe it was fate I'm headed back home. I deleted all my cookies last night
(made the mistake of shopping out of incognito mode so now all my web pages
are littered with advertisements for what I had been shopping for) and
github needs me to 2-factor auth. Apparently I left my phone recharging in
my car so I am locked out.

On Mon, Jan 30, 2017 at 9:51 AM, Eric Wiseblatt <ewiseblatt@google.com>
wrote:

> Thanks for catching the typos. I wish the client library had the ability
> to identify the requests once they've been created so I didnt need that
> duplicate string.
>
> I decided not to do the common tag map generator because there arguments
> vary among instances. Some are region, some are zone. Though the common
> method would only require 3 variables (api, region, zone) instead of 6, the
> zone and region are disjoint so looks more awkward given there arent many
> call sites.
>
> On Mon, Jan 30, 2017 at 8:55 AM, Matt Duftler <notifications@github.com>
> wrote:
>
>> *@duftler* approved this pull request.
>>
>> Thanks for the detailed explanation.
>> LGTM after addressing comments.
>> ------------------------------
>>
>> In clouddriver-google/src/main/groovy/com/netflix/spinnaker/clo
>> uddriver/google/deploy/ops/ModifyGoogleServerGroupInstanceTe
>> mplateAtomicOperation.groovy
>> <https://github.com/spinnaker/clouddriver/pull/1403#pullrequestreview-19077911>
>> :
>>
>> > @@ -103,16 +102,23 @@ class ModifyGoogleServerGroupInstanceTemplateAtomicOperation implements AtomicOp
>>      // Retrieve the managed instance group.
>>      def managedInstanceGroup =
>>        isRegional
>> -      ? instanceGroupManagers.get(project, region, serverGroupName).execute()
>> -      : instanceGroupManagers.get(project, zone, serverGroupName).execute()
>> +      ? timeExecute(instanceGroupManagers.get(project, region, serverGroupName),
>> +                    "compute.regionalInstanceGroupManagers",
>>
>> s/regional/region
>> ------------------------------
>>
>> In clouddriver-google/src/main/groovy/com/netflix/spinnaker/clo
>> uddriver/google/deploy/ops/ModifyGoogleServerGroupInstanceTe
>> mplateAtomicOperation.groovy
>> <https://github.com/spinnaker/clouddriver/pull/1403#pullrequestreview-19077911>
>> :
>>
>> > @@ -236,17 +245,23 @@ class ModifyGoogleServerGroupInstanceTemplateAtomicOperation implements AtomicOp
>>        if (isRegional) {
>>          def regionInstanceGroupManagersSetTemplateRequest =
>>            new RegionInstanceGroupManagersSetTemplateRequest(instanceTemplate: instanceTemplateUrl)
>> -        def setInstanceTemplateOperation = instanceGroupManagers.setInstanceTemplate(
>> -          project, region, serverGroupName, regionInstanceGroupManagersSetTemplateRequest).execute()
>> +        def setInstanceTemplateOperation = timeExecute(
>> +          instanceGroupManagers.setInstanceTemplate(
>> +            project, region, serverGroupName, regionInstanceGroupManagersSetTemplateRequest),
>> +          "compute.regionalInstanceGroupManagers.setInstanceTemplate",
>>
>> s/regional/region
>> ------------------------------
>>
>> In clouddriver-google/src/main/groovy/com/netflix/spinnaker/clo
>> uddriver/google/deploy/ops/UpsertGoogleServerGroupTagsAtomic
>> Operation.groovy
>> <https://github.com/spinnaker/clouddriver/pull/1403#pullrequestreview-19077911>
>> :
>>
>> > @@ -121,33 +132,46 @@ class UpsertGoogleServerGroupTagsAtomicOperation implements AtomicOperation<Void
>>      if (isRegional) {
>>        def regionInstanceGroupManagersSetTemplateRequest =
>>          new RegionInstanceGroupManagersSetTemplateRequest(instanceTemplate: instanceTemplateUrl)
>> -      def setInstanceTemplateOperation = instanceGroupManagers.setInstanceTemplate(
>> -        project, region, serverGroupName, regionInstanceGroupManagersSetTemplateRequest).execute()
>> +      def setInstanceTemplateOperation = timeExecute(
>> +              instanceGroupManagers.setInstanceTemplate(
>> +              project, region, serverGroupName, regionInstanceGroupManagersSetTemplateRequest),
>> +              "compute.regionalInstanceGroupManagers.setInstanceTemplate",
>>
>> s/regional/region
>> ------------------------------
>>
>> In clouddriver-google/src/main/groovy/com/netflix/spinnaker/clo
>> uddriver/google/deploy/ops/loadbalancer/DeleteGoogleSslLoadB
>> alancerAtomicOperation.groovy
>> <https://github.com/spinnaker/clouddriver/pull/1403#pullrequestreview-19077911>
>> :
>>
>> > @@ -142,7 +151,10 @@ class DeleteGoogleSslLoadBalancerAtomicOperation extends DeleteGoogleLoadBalance
>>      }
>>
>>      Operation deleteBackendServiceOp = GCEUtil.deleteIfNotInUse(
>> -      { compute.backendServices().delete(project, backendServiceName).execute() },
>> +      { timeExecute(
>> +            compute.backendServices().delete(project, backendServiceName),
>> +            "compute.backendSerfvices.delete",
>>
>> s/Serfvices/Services
>> ------------------------------
>>
>> In clouddriver-google/src/main/groovy/com/netflix/spinnaker/clo
>> uddriver/google/deploy/ops/loadbalancer/UpsertGoogleSslLoadB
>> alancerAtomicOperation.groovy
>> <https://github.com/spinnaker/clouddriver/pull/1403#pullrequestreview-19077911>
>> :
>>
>> > @@ -201,7 +222,10 @@ class UpsertGoogleSslLoadBalancerAtomicOperation extends UpsertGoogleLoadBalance
>>        existingBackendService.loadBalancingScheme = 'EXTERNAL'
>>        existingBackendService.protocol = description.ipProtocol
>>        backendServiceOp = safeRetry.doRetry(
>> -        { compute.backendServices().update(project, existingBackendService.getName(), existingBackendService).execute() },
>> +        { timeExecute(
>> +              compute.backendServices().update(project, existingBackendService.getName(), existingBackendService),
>> +              "compute.backendSerfvices.update",
>>
>> s/Serfvices/Services
>> ------------------------------
>>
>> In clouddriver-google/src/test/groovy/com/netflix/spinnaker/clo
>> uddriver/google/deploy/ops/DestroyGoogleServerGroupAtomicOpe
>> rationUnitSpec.groovy
>> <https://github.com/spinnaker/clouddriver/pull/1403#pullrequestreview-19077911>
>> :
>>
>> > @@ -234,6 +255,12 @@ class DestroyGoogleServerGroupAtomicOperationUnitSpec extends Specification {
>>          1 * computeMock.zoneOperations() >> zoneOperations
>>          1 * zoneOperations.get(PROJECT_NAME, location, INSTANCE_GROUP_OP_NAME) >> zoneOperationsGet
>>          1 * zoneOperationsGet.execute() >> instanceGroupManagersDeleteOp
>> +        registry.timer(
>> +            registry.createId("google.api",
>> +                  [api: "compute.instanceGroupManagers.delete",
>>
>> This map appears often enough in the tests that it might be worth it to
>> create a helper method for the tests, something like: Map
>> generateMetricsTagsMap(String api, String scope)
>>
>> —
>> You are receiving this because you authored the thread.
>> Reply to this email directly, view it on GitHub
>> <https://github.com/spinnaker/clouddriver/pull/1403#pullrequestreview-19077911>,
>> or mute the thread
>> <https://github.com/notifications/unsubscribe-auth/AEKkGHC_6X8JsWi4ikHZkBTUJSzTmiAAks5rXevmgaJpZM4Lwnbi>
>> .
>>
>
>
